common/environment/environment.cpp

```diff
@@ -31,11 +31,75 @@
 #include "dalienv.hpp"
 
 #define SDS_LOCK_TIMEOUT  30000
-
+#define DEFAULT_DROPZONE_INDEX      1
+#define DROPZONE_BY_MACHINE_SUFFIX  "-dropzoneByMachine-"
+#define DROPZONE_SUFFIX             "dropzone-"
+#define MACHINE_PREFIX              "machine-"
+#define NUMBER_OF_DROPZONES_SUFFIX  "-numberOfDropZones"
 
 static int environmentTraceLevel = 1;
 static Owned <IConstEnvironment> cache;
 
+class CLocalEnvironment;
+
+class CConstMachineInfoIterator : public CInterface, implements IConstMachineInfoIterator
```

Comment:
minor: CSimpleInterface preferred where CInterface baggage isn't required (e.g. beforeDispose/isAlive etc.)
minor(2): CSimpleInterfaceOf<IConstMachineInfoIterator> avoids a VMT and the IMPLEMENT_..